src/rgw/driver/sfs/multipart.cc

```diff
@@ -24,10 +24,14 @@ using namespace std;
 namespace rgw::sal {
 
 std::unique_ptr<rgw::sal::Object> SFSMultipartUpload::get_meta_obj() {
-  return std::make_unique<SFSMultipartMetaObject>(
-    store, rgw_obj_key(get_meta(), string(), RGW_OBJ_NS_MULTIPART),
-    bucket, bucketref
-  );
+  auto mmo = std::make_unique<SFSMultipartMetaObject>(
+      store, rgw_obj_key(get_meta(), string(), RGW_OBJ_NS_MULTIPART),
+      bucket, bucketref
+    );
```

Comment:
@0xavi0 @jecluis @irq0 these following three lines should represent the fix for the problem we were debugging.
There is no need to implement complete persistence for attributes in the multipart upload context (for this PR to work).
`get_meta_obj()` needed to be updated to reflect the attributes to the returned `mmo` obj.

Reply:
lgtm, but this should be in a different commit. It's a different logical change than the rest of the patch. I would suggest spinning this code to a new patch and have it before the remaining code changes in the git history.